dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -0,0 +1,537 @@
+                    .setConnectionTimeToLive(TimeValue.ofMilliseconds(idleConnectionTimeoutMillis))
+                    .setDefaultSocketConfig(SocketConfig.custom()
+                            .setSoKeepAlive(true)
+                            .setSoTimeout(Timeout.of(socketTimeoutMillis, TimeUnit.MILLISECONDS))
+                            .build())
+                    .setMaxConnPerRoute(Integer.MAX_VALUE)
+                    .setMaxConnTotal(Integer.MAX_VALUE)
+                    .setValidateAfterInactivity(connectionPoolInactivityCheck)
+                    .build();
+
+            setupConnectionPoolMetrics(conf.taggedMetricRegistry(), name, connectionManager);
+
+            HttpClientBuilder builder = HttpClients.custom()
+                    .setDefaultRequestConfig(RequestConfig.custom()
+                            .setConnectTimeout(connectTimeout)
+                            // Don't allow clients to block forever waiting on a connection to become available
+                            .setConnectionRequestTimeout(connectTimeout)
+                            // Match okhttp, disallow redirects
+                            .setRedirectsEnabled(false)
+                            .setAuthenticationEnabled(conf.proxyCredentials().isPresent())
+                            .setExpectContinueEnabled(false)
+                            .setDefaultKeepAlive(idleConnectionTimeoutMillis, TimeUnit.MILLISECONDS)
+                            .build())
+                    // Connection pool lifecycle must be managed separately. This allows us to configure a more
+                    // precise IdleConnectionEvictor.
+                    .setConnectionManagerShared(true)
+                    .setConnectionManager(connectionManager)
+                    .setRoutePlanner(new SystemDefaultRoutePlanner(null, conf.proxy()))
+                    .disableAutomaticRetries()
+                    // Must be disabled otherwise connections are not reused when client certificates are provided
+                    .disableConnectionState()
+                    // Match okhttp behavior disabling cookies
+                    .disableCookieManagement()
+                    // Dialogue handles content-compression with ContentDecodingChannel
+                    .disableContentCompression()
+                    .setDefaultCredentialsProvider(NullCredentialsProvider.INSTANCE)
+                    .setTargetAuthenticationStrategy(NullAuthenticationStrategy.INSTANCE)
+                    .setProxyAuthenticationStrategy(NullAuthenticationStrategy.INSTANCE)
+                    .setDefaultAuthSchemeRegistry(
+                            RegistryBuilder.<AuthSchemeFactory>create().build());
+            conf.proxyCredentials().ifPresent(credentials -> builder.setDefaultCredentialsProvider(
+                            new SingleCredentialsProvider(credentials))
+                    .setProxyAuthenticationStrategy(DefaultAuthenticationStrategy.INSTANCE)
```

Comment:
There's no more dedicated ProxyAuthenticator. There's only default from what I could find and it handles proxy authentication